app/assets/javascripts/Components/assignment_chart.jsx

```diff
@@ -103,84 +103,114 @@ export class AssignmentChart extends React.Component {
+      } else {
+        return <span>{stat_label}</span>;
+      }
+    };
+
+    const refresh_stats_link = (
+      <React.Fragment>
+        <span />
+        <a
+          data-remote="true"
+          href={Routes.view_summary_course_assignment_path(
+            this.props.course_id,
+            this.props.assessment_id
+          )}
+        >
+          {I18n.t("refresh")}
+        </a>
+      </React.Fragment>
+    );
+
+    const assignment_summary_stats = (
+      <div className={this.props.is_expanded ? "assignment-summary-stats-info" : "flex-row-expand"}>
+        <div className="grid-2-col">
+          {renderStatLabel(I18n.t("average"))}
+          <span>{(this.state.summary.average || 0).toFixed(2)}%</span>
+          {renderStatLabel(I18n.t("median"))}
+          <span>{(this.state.summary.median || 0).toFixed(2)}%</span>
+          {renderStatLabel(I18n.t("assignments_submitted"))}
+          <span>
+            {this.state.summary.num_submissions_collected} / {this.state.summary.groupings_size}
+          </span>
+          {renderStatLabel(I18n.t("assignments_graded"))}
+          <span>
+            {this.state.summary.num_submissions_graded} / {this.state.summary.groupings_size}
+          </span>
+          {renderStatLabel(I18n.t("num_failed"))}
+          <span>{this.state.summary.num_fails}</span>
+          {renderStatLabel(I18n.t("num_zeros"))}
+          <span>{this.state.summary.num_zeros}</span>
+          {this.props.is_expanded ? refresh_stats_link : ""}
+        </div>
+        {outstanding_remark_request_link}
+        <p>{!this.props.is_expanded ? refresh_stats_link : ""}</p>
+      </div>
+    );
+
+    let assignment_graph_header = (
+      <h2>
+        <a
+          href={Routes.browse_course_assignment_submissions_path(
+            this.props.course_id,
+            this.props.assessment_id
+          )}
+        >
+          {this.state.summary.name}
+        </a>
+      </h2>
+    );
+    if (this.props.is_expanded) {
+      assignment_graph_header = <h3>{I18n.t("assignment_distribution")}</h3>;
+    }
+
     const assignment_graph = (
       <React.Fragment>
-        <h2>
-          <a
-            href={Routes.browse_course_assignment_submissions_path(
-              this.props.course_id,
-              this.props.assessment_id
-            )}
-          >
-            {this.state.summary.name}
-          </a>
-        </h2>
-        <div className="flex-row">
+        <div>
+          {assignment_graph_header}
+          <div className={this.props.is_expanded ? "" : "flex-row"}>
+            <div>
+              <Bar
+                data={this.state.assignment_grade_distribution.data}
+                options={this.state.assignment_grade_distribution.options}
+                width={this.props.is_expanded ? "" : "500"}
+                height={this.props.is_expanded ? "200vh" : "450"}
+              />
+            </div>
+            {!this.props.is_expanded ? assignment_summary_stats : ""}
```

Comment:
Similar to above comment, the HTML structure should be the same in both views. You shouldn't need to reposition elements. The overall layout is the same.